ee/clickhouse/queries/trends/total_volume.py

```diff
@@ -38,48 +33,156 @@ def _total_volume_query(self, entity: Entity, filter: Filter, team_id: int) -> T
+                **trend_event_query.active_user_params
             )
         else:
+            final_query = VOLUME_SQL.format(event_query=event_query, **content_sql_params)
 
-            if entity.math in [WEEKLY_ACTIVE, MONTHLY_ACTIVE]:
-                content_sql = ACTIVE_USER_SQL.format(
-                    event_query=event_query,
-                    **content_sql_params,
-                    parsed_date_to=trend_event_query.parsed_date_to,
-                    parsed_date_from=trend_event_query.parsed_date_from,
-                    **trend_event_query.active_user_params
-                )
-            else:
-                content_sql = VOLUME_SQL.format(event_query=event_query, **content_sql_params)
-
-            null_sql = NULL_SQL.format(trunc_func=trunc_func, interval_func=interval_func)
-            params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
-            params["date_to"] = format_ch_timestamp(filter.date_to, filter)
-            params["interval"] = filter.interval
-            final_query = AGGREGATE_SQL.format(null_sql=null_sql, content_sql=content_sql)
-            return final_query, params, self._parse_total_volume_result(filter)
+        params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
+        params["date_to"] = format_ch_timestamp(filter.date_to, filter)
+        params["interval"] = filter.interval
+        return final_query, params, self._parse_total_volume_result(filter)
 
     def _parse_total_volume_result(self, filter: Filter) -> Callable:
         def _parse(result: List) -> List:
-            parsed_results = []
-            for _, stats in enumerate(result):
-                parsed_result = parse_response(stats, filter)
-                parsed_results.append(parsed_result)
+            # NOTE: there's some implicit coupling here:
+            #
+            #        1. the ordering of the fields we get back, (date, value)
+            #        2. that the last row is always the `WITH TOTALS` result
+            #
+            #  TODO: remove coupling
+
+            #  The last row relates to the `WITH TOTALS` from the clickhouse query
+            values_without_totals, (_, aggregated_value) = result[:-1], result[-1]
+
+            # Separate dates and values, as desired for the REST API response
+            dates, values = zip(*sorted(values_without_totals))
+
+            parsed_result = parse_response([dates, values], filter)
 
-            return parsed_results
+            # NOTE: to maintain backwards compatability here from previous behaviour, we exclude the
+            #        `aggregated_value` for some filter.display.
+            # TODO: verify if updating to always including `aggregated_value` would break anything. It
+            #       would be simpler to not have to have this special casing in here.
+            if filter.display in TRENDS_DISPLAY_BY_VALUE:
```

Comment:
Why isn't this done in parsing the result?

Reply:
Yup I can move it there.